src/components/grid.jsx

```diff
@@ -47,28 +47,19 @@ function Grid({
     }
   };
 
-  // fixme after markup changes
-  const columnsCount =
-    images.length === 12 ? 3 : 4;
-
   return (
     <div className="container">
       <div className="row no-gutters">
         {images.map((item, index) => (
           <Card
             key={item.id}
             item={item}
-            visible={visibleItems.includes(
+            isVisible={visibleItems.includes(
               index
             )}
-            finished={finishedItems.includes(
+            isFinished={finishedItems.includes(
               index
             )}
-            className={`${
-              columnsCount === 3
-                ? 'col-4'
-                : 'col-3'
-            }`}
             onCardClick={() =>
               handleCardClick(index)
             }
```

Comment:
Имхо, можно сделать изящнее:
```suggestion
            onCardClick={handleCardClick}
```
А карточка, вызывая функцию, будет передавать собственный айдишник. Может, это вкусовщина, но мне не нравится, когда логика зависит от индекса. Тем более, что айдишники у нас есть!